dask/dataframe/core.py

```diff
@@ -300,6 +290,33 @@ def _elemwise(self):
         return elemwise
 
     @property
+    def _repr_data(self):
+        raise NotImplementedError
+
+    @property
+    def _repr_name(self):
+        return self._name if len(self._name) < 10 else self._name[:7]
```

Comment:
If we wanted to use more than 7 or 10 characters then we might want to use the `dask.utils.key_split` function introduced in https://github.com/dask/dask/pull/1919 .